components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/actions/ExtendWorkspaceTimeoutAction.kt

```diff
@@ -11,6 +11,64 @@ import com.intellij.openapi.diagnostic.thisLogger
 import io.gitpod.gitpodprotocol.api.entities.WorkspaceTimeoutDuration
 import io.gitpod.jetbrains.remote.GitpodManager
 import com.intellij.notification.NotificationType
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import javax.swing.JComponent
+import javax.swing.JTextField
+import javax.swing.JPanel
+import javax.swing.JComboBox
+import javax.swing.BoxLayout
+
+// validation from https://github.com/gitpod-io/gitpod/blob/74ccaea38db8df2d1666161a073015485ebb90ca/components/gitpod-protocol/src/gitpod-service.ts#L361-L383
+const val WORKSPACE_MAXIMUM_TIMEOUT_HOURS = 24
+fun validate(duration: Int, unit: Char): String {
+    if (duration <= 0) {
+        throw IllegalArgumentException("Invalid timeout value: ${duration}${unit}")
+    }
+    if (
+        (unit == 'h' && duration > WORKSPACE_MAXIMUM_TIMEOUT_HOURS) ||
+        (unit == 'm' && duration > WORKSPACE_MAXIMUM_TIMEOUT_HOURS * 60)
+    ) {
+        throw IllegalArgumentException("Timeouts cannot extend to more than a day")
```

Comment:
[as above](https://github.com/gitpod-io/gitpod/pull/16267/files#r1101853983)

```suggestion
        throw IllegalArgumentException("Workspace inactivity timeout cannot exceed 24h")
```